{
    "version": "https://jsonfeed.org/version/1",
    "title": "Security Advisory for Python packages hosted at PyPI.org",
    "home_page_url": "https://github.com/advisories?query=type%3Areviewed+ecosystem%3Apip",
    "feed_url": "https://azu.github.io/github-advisory-database-rss/pip.json",
    "description": "Security Advisory for Python packages hosted at PyPI.org on GitHub",
    "items": [
        {
            "content_html": "<p>Pillow before 9.2.0 performs Improper Handling of Highly Compressed GIF Data (Data Amplification).</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-45198\">https://nvd.nist.gov/vuln/detail/CVE-2022-45198</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/6402\">https://github.com/python-pillow/Pillow/pull/6402</a></li>\n<li><a href=\"https://bugs.gentoo.org/855683\">https://bugs.gentoo.org/855683</a></li>\n<li><a href=\"https://cwe.mitre.org/data/definitions/409.html\">https://cwe.mitre.org/data/definitions/409.html</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/releases/tag/9.2.0\">https://github.com/python-pillow/Pillow/releases/tag/9.2.0</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/6402/commits/c9f1b35e981075110a23487a8d4a6cbb59a588ea\">https://github.com/python-pillow/Pillow/pull/6402/commits/c9f1b35e981075110a23487a8d4a6cbb59a588ea</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4\">https://github.com/python-pillow/Pillow/commit/11918eac0628ec8ac0812670d9838361ead2d6a4</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/202211-10\">https://security.gentoo.org/glsa/202211-10</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-m2vv-5vj5-2hm7\">https://github.com/advisories/GHSA-m2vv-5vj5-2hm7</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-m2vv-5vj5-2hm7",
            "title": "[pillow] Pillow vulnerable to Data Amplification attack.",
            "date_modified": "2023-09-05T09:35:58.000Z",
            "date_published": "2022-11-14T12:00:15.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-m2vv-5vj5-2hm7"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The package joblib from 0 and before 1.2.0 is vulnerable to Arbitrary Code Execution via the <code>pre_dispatch</code> flag in <code>Parallel()</code> class due to the <code>eval()</code> statement.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2022-21797\">https://nvd.nist.gov/vuln/detail/CVE-2022-21797</a></li>\n<li><a href=\"https://github.com/joblib/joblib/issues/1128\">https://github.com/joblib/joblib/issues/1128</a></li>\n<li><a href=\"https://github.com/joblib/joblib/pull/1321\">https://github.com/joblib/joblib/pull/1321</a></li>\n<li><a href=\"https://github.com/joblib/joblib/commit/b90f10efeb670a2cc877fb88ebb3f2019189e059\">https://github.com/joblib/joblib/commit/b90f10efeb670a2cc877fb88ebb3f2019189e059</a></li>\n<li><a href=\"https://security.snyk.io/vuln/SNYK-PYTHON-JOBLIB-3027033\">https://security.snyk.io/vuln/SNYK-PYTHON-JOBLIB-3027033</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BVOMMW37OXZWU2EV5ONAAS462IQEHZOF/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BVOMMW37OXZWU2EV5ONAAS462IQEHZOF/</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2022/11/msg00020.html\">https://lists.debian.org/debian-lts-announce/2022/11/msg00020.html</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MJ5XTJS6OKJRRVXWFN5J67K3BYPEOBDF/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/MJ5XTJS6OKJRRVXWFN5J67K3BYPEOBDF/</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2023/03/msg00027.html\">https://lists.debian.org/debian-lts-announce/2023/03/msg00027.html</a></li>\n<li><a href=\"https://github.com/joblib/joblib/pull/1327\">https://github.com/joblib/joblib/pull/1327</a></li>\n<li><a href=\"https://github.com/joblib/joblib/pull/1352\">https://github.com/joblib/joblib/pull/1352</a></li>\n<li><a href=\"https://github.com/joblib/joblib/commit/6638b9e9711ad1ebbf6dd95aa7cee0dca9897b42\">https://github.com/joblib/joblib/commit/6638b9e9711ad1ebbf6dd95aa7cee0dca9897b42</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/joblib/PYSEC-2022-288.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/joblib/PYSEC-2022-288.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-6hrg-qmvc-2xh8\">https://github.com/advisories/GHSA-6hrg-qmvc-2xh8</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-6hrg-qmvc-2xh8",
            "title": "[joblib] joblib vulnerable to arbitrary code execution",
            "date_modified": "2023-09-05T09:35:28.000Z",
            "date_published": "2022-09-27T00:00:22.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-6hrg-qmvc-2xh8"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>rpc.py through 0.6.0 allows Remote Code Execution because an unpickle occurs when the \"serializer: pickle\" HTTP header is sent. In other words, although JSON (not Pickle) is the default data format, an unauthenticated client can cause the data to be processed with unpickle.</p>\n<p><a href=\"https://github.com/abersheeran/rpc.py/issues/22\">Per the maintainer</a>, rpc.py is not designed for an API that is open to the outside world, and external requests cannot reach rpc.py in real world use.</p>\n<p>A <a href=\"https://github.com/abersheeran/rpc.py/commit/491e7a841ed9a754796d6ab047a9fb16e23bf8bd\">fix</a> exists on the <code>master</code> branch. As a workaround, use the following code to turn off pickle in older versions:</p>\n<pre><code>del SERIALIZER_NAMES[PickleSerializer.name]\ndel SERIALIZER_TYPES[PickleSerializer.content_type]\n### References\n\n- &lt;https://nvd.nist.gov/vuln/detail/CVE-2022-35411&gt;\n- &lt;https://github.com/abersheeran/rpc.py/commit/491e7a841ed9a754796d6ab047a9fb16e23bf8bd&gt;\n- &lt;https://github.com/ehtec/rpcpy-exploit&gt;\n- &lt;https://medium.com/@elias.hohl/remote-code-execution-0-day-in-rpc-py-709c76690c30&gt;\n- &lt;https://github.com/abersheeran/rpc.py/issues/22&gt;\n- &lt;http://packetstormsecurity.com/files/167872/rpc.py-0.6.0-Remote-Code-Execution.html&gt;\n- &lt;https://github.com/advisories/GHSA-8rq8-f485-7v8x&gt;\n</code></pre>\n",
            "url": "https://github.com/advisories/GHSA-8rq8-f485-7v8x",
            "title": "[rpc.py] rpc.py vulnerable to Deserialization of Untrusted Data",
            "date_modified": "2023-09-05T09:35:06.000Z",
            "date_published": "2022-07-09T00:00:19.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8rq8-f485-7v8x"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in Project Jupyter JupyterHub OAuthenticator 0.6.x before 0.6.2 and 0.7.x before 0.7.3. When using JupyterHub with GitLab group whitelisting for access control, group membership was not checked correctly, allowing members not in the whitelisted groups to create accounts on the Hub. (Users were not allowed to access other users' accounts, but could create their own accounts on the Hub linked to their GitLab account. GitLab authentication not using gitlab_group_whitelist is unaffected. No other Authenticators are affected.)</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-7206\">https://nvd.nist.gov/vuln/detail/CVE-2018-7206</a></li>\n<li><a href=\"https://github.com/jupyterhub/oauthenticator/commit/1845c0e4b1bff3462c91c3108c85205acd3c75a2\">https://github.com/jupyterhub/oauthenticator/commit/1845c0e4b1bff3462c91c3108c85205acd3c75a2</a></li>\n<li><a href=\"https://blog.jupyter.org/security-fix-for-jupyterhub-gitlab-oauthenticator-7b14571d1f76\">https://blog.jupyter.org/security-fix-for-jupyterhub-gitlab-oauthenticator-7b14571d1f76</a></li>\n<li><a href=\"https://github.com/jupyterhub/oauthenticator/blob/8499dc2/CHANGELOG.md#073---2018-02-16\">https://github.com/jupyterhub/oauthenticator/blob/8499dc2/CHANGELOG.md#073---2018-02-16</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/oauthenticator/PYSEC-2018-68.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/oauthenticator/PYSEC-2018-68.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8x3m-m3x9-54fj\">https://github.com/advisories/GHSA-8x3m-m3x9-54fj</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8x3m-m3x9-54fj",
            "title": "[oauthenticator] JupyterHub OAuthenticator elevation of privilege",
            "date_modified": "2023-09-05T09:32:16.000Z",
            "date_published": "2022-05-13T01:12:09.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8x3m-m3x9-54fj"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>An issue was discovered in Project Jupyter JupyterHub OAuthenticator 0.6.x before 0.6.2 and 0.7.x before 0.7.3. When using JupyterHub with GitLab group whitelisting for access control, group membership was not checked correctly, allowing members not in the whitelisted groups to create accounts on the Hub. (Users were not allowed to access other users' accounts, but could create their own accounts on the Hub linked to their GitLab account. GitLab authentication not using gitlab_group_whitelist is unaffected. No other Authenticators are affected.)</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2018-7206\">https://nvd.nist.gov/vuln/detail/CVE-2018-7206</a></li>\n<li><a href=\"https://github.com/jupyterhub/oauthenticator/commit/1845c0e4b1bff3462c91c3108c85205acd3c75a2\">https://github.com/jupyterhub/oauthenticator/commit/1845c0e4b1bff3462c91c3108c85205acd3c75a2</a></li>\n<li><a href=\"https://blog.jupyter.org/security-fix-for-jupyterhub-gitlab-oauthenticator-7b14571d1f76\">https://blog.jupyter.org/security-fix-for-jupyterhub-gitlab-oauthenticator-7b14571d1f76</a></li>\n<li><a href=\"https://github.com/jupyterhub/oauthenticator/blob/8499dc2/CHANGELOG.md#073---2018-02-16\">https://github.com/jupyterhub/oauthenticator/blob/8499dc2/CHANGELOG.md#073---2018-02-16</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/oauthenticator/PYSEC-2018-68.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/oauthenticator/PYSEC-2018-68.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8x3m-m3x9-54fj\">https://github.com/advisories/GHSA-8x3m-m3x9-54fj</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8x3m-m3x9-54fj",
            "title": "[oauthenticator] JupyterHub OAuthenticator elevation of privilege",
            "date_modified": "2023-09-05T09:32:16.000Z",
            "date_published": "2022-05-13T01:12:09.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8x3m-m3x9-54fj"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In Pylons Colander through 1.6, the URL validator allows an attacker to potentially cause an infinite loop thereby causing a denial of service via an unclosed parenthesis.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2017-18361\">https://nvd.nist.gov/vuln/detail/CVE-2017-18361</a></li>\n<li><a href=\"https://github.com/Pylons/colander/issues/290\">https://github.com/Pylons/colander/issues/290</a></li>\n<li><a href=\"https://github.com/Pylons/colander/pull/323\">https://github.com/Pylons/colander/pull/323</a></li>\n<li><a href=\"https://github.com/Pylons/colander/commit/98805557c10ab5ff3016ed09aa2d48c49b9df40b\">https://github.com/Pylons/colander/commit/98805557c10ab5ff3016ed09aa2d48c49b9df40b</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rv95-4wxj-6fqq\">https://github.com/advisories/GHSA-rv95-4wxj-6fqq</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rv95-4wxj-6fqq",
            "title": "[colander] Pylons Colander Denial of Service vulnerability",
            "date_modified": "2023-09-05T09:29:43.000Z",
            "date_published": "2019-02-07T18:18:22.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rv95-4wxj-6fqq"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Integer overflow in the <code>ImagingResampleHorizontal</code> function in <code>libImaging/Resample.c</code> in Pillow before 3.1.1 allows remote attackers to have unspecified impact via negative values of the new size, which triggers a heap-based buffer overflow.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2016-4009\">https://nvd.nist.gov/vuln/detail/CVE-2016-4009</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/1714\">https://github.com/python-pillow/Pillow/pull/1714</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/4e0d9b0b9740d258ade40cce248c93777362ac1e\">https://github.com/python-pillow/Pillow/commit/4e0d9b0b9740d258ade40cce248c93777362ac1e</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-hvr8-466p-75rh\">https://github.com/advisories/GHSA-hvr8-466p-75rh</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/blob/c3cb690fed5d4bf0c45576759de55d054916c165/CHANGES.rst\">https://github.com/python-pillow/Pillow/blob/c3cb690fed5d4bf0c45576759de55d054916c165/CHANGES.rst</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/201612-52\">https://security.gentoo.org/glsa/201612-52</a></li>\n<li><a href=\"http://www.securityfocus.com/bid/86064\">http://www.securityfocus.com/bid/86064</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-hvr8-466p-75rh",
            "title": "[Pillow] Pillow Integer overflow in ImagingResampleHorizontal",
            "date_modified": "2023-09-05T09:28:31.000Z",
            "date_published": "2018-07-24T20:15:48.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-hvr8-466p-75rh"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p><code>libImaging/FliDecode.c</code> in Pillow before 6.2.2 has an FLI buffer overflow.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-5313\">https://nvd.nist.gov/vuln/detail/CVE-2020-5313</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/a09acd0decd8a87ccce939d5ff65dab59e7d365b\">https://github.com/python-pillow/Pillow/commit/a09acd0decd8a87ccce939d5ff65dab59e7d365b</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/blob/master/CHANGES.rst#622-2020-01-02\">https://github.com/python-pillow/Pillow/blob/master/CHANGES.rst#622-2020-01-02</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2MMU3WT2X64GS5WHDPKKC2WZA7UIIQ3A/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2MMU3WT2X64GS5WHDPKKC2WZA7UIIQ3A/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/3DUMIBUYGJRAVJCTFUWBRLVQKOUTVX5P/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/3DUMIBUYGJRAVJCTFUWBRLVQKOUTVX5P/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/6.2.2.html\">https://pillow.readthedocs.io/en/stable/releasenotes/6.2.2.html</a></li>\n<li><a href=\"https://usn.ubuntu.com/4272-1/\">https://usn.ubuntu.com/4272-1/</a></li>\n<li><a href=\"https://www.debian.org/security/2020/dsa-4631\">https://www.debian.org/security/2020/dsa-4631</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-hj69-c76v-86wr\">https://github.com/advisories/GHSA-hj69-c76v-86wr</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-hj69-c76v-86wr",
            "title": "[Pillow] Out-of-bounds Read in Pillow",
            "date_modified": "2023-09-05T09:26:38.000Z",
            "date_published": "2020-04-01T16:36:00.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-hj69-c76v-86wr"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The (1) JpegImagePlugin.py and (2) EpsImagePlugin.py scripts in Python Image Library (PIL) 1.1.7 and earlier and Pillow before 2.3.1 uses the names of temporary files on the command line, which makes it easier for local users to conduct symlink attacks by listing the processes.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2014-1933\">https://nvd.nist.gov/vuln/detail/CVE-2014-1933</a></li>\n<li><a href=\"https://github.com/python-imaging/Pillow/commit/4e9f367dfd3f04c8f5d23f7f759ec12782e10ee7\">https://github.com/python-imaging/Pillow/commit/4e9f367dfd3f04c8f5d23f7f759ec12782e10ee7</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/201612-52\">https://security.gentoo.org/glsa/201612-52</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-updates/2014-05/msg00002.html\">http://lists.opensuse.org/opensuse-updates/2014-05/msg00002.html</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2014/02/10/15\">http://www.openwall.com/lists/oss-security/2014/02/10/15</a></li>\n<li><a href=\"http://www.openwall.com/lists/oss-security/2014/02/11/1\">http://www.openwall.com/lists/oss-security/2014/02/11/1</a></li>\n<li><a href=\"http://www.securityfocus.com/bid/65513\">http://www.securityfocus.com/bid/65513</a></li>\n<li><a href=\"http://www.ubuntu.com/usn/USN-2168-1\">http://www.ubuntu.com/usn/USN-2168-1</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-r854-96gq-rfg3\">https://github.com/advisories/GHSA-r854-96gq-rfg3</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-r854-96gq-rfg3",
            "title": "[Pillow] Pillow Temporary file name leakage",
            "date_modified": "2023-09-05T09:25:58.000Z",
            "date_published": "2020-05-18T17:41:19.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-r854-96gq-rfg3"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Buffer overflow in the <code>ImagingFliDecode</code> function in <code>libImaging/FliDecode.c</code> in Pillow before 3.1.1 allows remote attackers to cause a denial of service (crash) via a crafted FLI file.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2016-0775\">https://nvd.nist.gov/vuln/detail/CVE-2016-0775</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/893a40850c2d5da41537958e40569c029a6e127b\">https://github.com/python-pillow/Pillow/commit/893a40850c2d5da41537958e40569c029a6e127b</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8xjv-v9xq-m5h9\">https://github.com/advisories/GHSA-8xjv-v9xq-m5h9</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/blob/c3cb690fed5d4bf0c45576759de55d054916c165/CHANGES.rst\">https://github.com/python-pillow/Pillow/blob/c3cb690fed5d4bf0c45576759de55d054916c165/CHANGES.rst</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/201612-52\">https://security.gentoo.org/glsa/201612-52</a></li>\n<li><a href=\"http://www.debian.org/security/2016/dsa-3499\">http://www.debian.org/security/2016/dsa-3499</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8xjv-v9xq-m5h9",
            "title": "[Pillow] Pillow Buffer overflow in ImagingFliDecode",
            "date_modified": "2023-09-05T09:23:17.000Z",
            "date_published": "2018-07-24T20:15:36.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8xjv-v9xq-m5h9"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Buffer overflow in the <code>ImagingLibTiffDecode</code> function in <code>libImaging/TiffDecode.c</code> in Pillow before 3.1.1 allows remote attackers to overwrite memory via a crafted TIFF file.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2016-0740\">https://nvd.nist.gov/vuln/detail/CVE-2016-0740</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/6dcbf5bd96b717c58d7b642949da8d323099928e\">https://github.com/python-pillow/Pillow/commit/6dcbf5bd96b717c58d7b642949da8d323099928e</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-hggx-3h72-49ww\">https://github.com/advisories/GHSA-hggx-3h72-49ww</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/blob/c3cb690fed5d4bf0c45576759de55d054916c165/CHANGES.rst\">https://github.com/python-pillow/Pillow/blob/c3cb690fed5d4bf0c45576759de55d054916c165/CHANGES.rst</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/201612-52\">https://security.gentoo.org/glsa/201612-52</a></li>\n<li><a href=\"http://www.debian.org/security/2016/dsa-3499\">http://www.debian.org/security/2016/dsa-3499</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-hggx-3h72-49ww",
            "title": "[Pillow] Pillow Buffer overflow in ImagingLibTiffDecode",
            "date_modified": "2023-09-05T09:22:11.000Z",
            "date_published": "2018-07-24T20:03:51.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-hggx-3h72-49ww"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Pillow before 3.3.2 allows context-dependent attackers to obtain sensitive information by using the \"crafted image file\" approach, related to an \"Integer Overflow\" issue affecting the <code>Image.core.map_buffer</code> in <code>map.c</code> component.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2016-9189\">https://nvd.nist.gov/vuln/detail/CVE-2016-9189</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/issues/2105\">https://github.com/python-pillow/Pillow/issues/2105</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/2146/commits/c50ebe6459a131a1ea8ca531f10da616d3ceaa0f\">https://github.com/python-pillow/Pillow/pull/2146/commits/c50ebe6459a131a1ea8ca531f10da616d3ceaa0f</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-rwr3-c2q8-gm56\">https://github.com/advisories/GHSA-rwr3-c2q8-gm56</a></li>\n<li><a href=\"https://security.gentoo.org/glsa/201612-52\">https://security.gentoo.org/glsa/201612-52</a></li>\n<li><a href=\"http://pillow.readthedocs.io/en/3.4.x/releasenotes/3.3.2.html\">http://pillow.readthedocs.io/en/3.4.x/releasenotes/3.3.2.html</a></li>\n<li><a href=\"http://www.debian.org/security/2016/dsa-3710\">http://www.debian.org/security/2016/dsa-3710</a></li>\n<li><a href=\"http://www.securityfocus.com/bid/94234\">http://www.securityfocus.com/bid/94234</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-rwr3-c2q8-gm56",
            "title": "[Pillow] Pillow Integer overflow in Map.c",
            "date_modified": "2023-09-05T09:20:58.000Z",
            "date_published": "2018-07-24T20:08:27.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-rwr3-c2q8-gm56"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Pillow before 7.1.0 has multiple out-of-bounds reads in <code>libImaging/FliDecode.c</code>.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10177\">https://nvd.nist.gov/vuln/detail/CVE-2020-10177</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/4503\">https://github.com/python-pillow/Pillow/pull/4503</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/4538\">https://github.com/python-pillow/Pillow/pull/4538</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commits/master/src/libImaging\">https://github.com/python-pillow/Pillow/commits/master/src/libImaging</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/6.2.3.html\">https://pillow.readthedocs.io/en/stable/releasenotes/6.2.3.html</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html\">https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html</a></li>\n<li><a href=\"https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574573\">https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574573</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-1/\">https://usn.ubuntu.com/4430-1/</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-2/\">https://usn.ubuntu.com/4430-2/</a></li>\n<li><a href=\"https://lists.debian.org/debian-lts-announce/2020/08/msg00012.html\">https://lists.debian.org/debian-lts-announce/2020/08/msg00012.html</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/issues/4750\">https://github.com/python-pillow/Pillow/issues/4750</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-cqhg-xjhh-p8hf\">https://github.com/advisories/GHSA-cqhg-xjhh-p8hf</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-cqhg-xjhh-p8hf",
            "title": "[Pillow] Out-of-bounds reads in Pillow",
            "date_modified": "2023-09-05T09:19:25.000Z",
            "date_published": "2020-07-27T21:52:43.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-cqhg-xjhh-p8hf"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p><code>libImaging/SgiRleDecode.c</code> in Pillow before 6.2.2 has an SGI buffer overflow.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-5311\">https://nvd.nist.gov/vuln/detail/CVE-2020-5311</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/a79b65c47c7dc6fe623aadf09aa6192fc54548f3\">https://github.com/python-pillow/Pillow/commit/a79b65c47c7dc6fe623aadf09aa6192fc54548f3</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2020:0566\">https://access.redhat.com/errata/RHSA-2020:0566</a></li>\n<li><a href=\"https://access.redhat.com/errata/RHSA-2020:0580\">https://access.redhat.com/errata/RHSA-2020:0580</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2MMU3WT2X64GS5WHDPKKC2WZA7UIIQ3A/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/2MMU3WT2X64GS5WHDPKKC2WZA7UIIQ3A/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/3DUMIBUYGJRAVJCTFUWBRLVQKOUTVX5P/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/3DUMIBUYGJRAVJCTFUWBRLVQKOUTVX5P/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/6.2.2.html\">https://pillow.readthedocs.io/en/stable/releasenotes/6.2.2.html</a></li>\n<li><a href=\"https://usn.ubuntu.com/4272-1/\">https://usn.ubuntu.com/4272-1/</a></li>\n<li><a href=\"https://www.debian.org/security/2020/dsa-4631\">https://www.debian.org/security/2020/dsa-4631</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-r7rm-8j6h-r933\">https://github.com/advisories/GHSA-r7rm-8j6h-r933</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2020-82.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/pillow/PYSEC-2020-82.yaml</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-r7rm-8j6h-r933",
            "title": "[pillow] Buffer Copy without Checking Size of Input in Pillow",
            "date_modified": "2023-09-05T09:18:50.000Z",
            "date_published": "2022-05-24T17:05:33.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-r7rm-8j6h-r933"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In <code>libImaging/PcxDecode.c</code> in Pillow before 7.1.0, an out-of-bounds read can occur when reading PCX files where <code>state-&gt;shuffle</code> is instructed to read beyond <code>state-&gt;buffer</code>.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10378\">https://nvd.nist.gov/vuln/detail/CVE-2020-10378</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/4538\">https://github.com/python-pillow/Pillow/pull/4538</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/6a83e4324738bb0452fbe8074a995b1c73f08de7#diff-9478f2787e3ae9668a15123b165c23ac\">https://github.com/python-pillow/Pillow/commit/6a83e4324738bb0452fbe8074a995b1c73f08de7#diff-9478f2787e3ae9668a15123b165c23ac</a></li>\n<li><a href=\"https://github.com/pypa/advisory-db/blob/7872b0a91b4d980f749e6d75a81f8cc1af32829f/vulns/pillow/PYSEC-2020-77.yaml\">https://github.com/pypa/advisory-db/blob/7872b0a91b4d980f749e6d75a81f8cc1af32829f/vulns/pillow/PYSEC-2020-77.yaml</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commits/master/src/libImaging\">https://github.com/python-pillow/Pillow/commits/master/src/libImaging</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/6.2.3.html\">https://pillow.readthedocs.io/en/stable/releasenotes/6.2.3.html</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html\">https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-1/\">https://usn.ubuntu.com/4430-1/</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-2/\">https://usn.ubuntu.com/4430-2/</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/issues/4750\">https://github.com/python-pillow/Pillow/issues/4750</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-3xv8-3j54-hgrp\">https://github.com/advisories/GHSA-3xv8-3j54-hgrp</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-3xv8-3j54-hgrp",
            "title": "[Pillow] Out-of-bounds read in Pillow",
            "date_modified": "2023-09-05T09:18:07.000Z",
            "date_published": "2021-11-03T18:04:53.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-3xv8-3j54-hgrp"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In Pillow before 7.1.0, there are two Buffer Overflows in <code>libImaging/TiffDecode.c</code>.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10379\">https://nvd.nist.gov/vuln/detail/CVE-2020-10379</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/4538\">https://github.com/python-pillow/Pillow/pull/4538</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/46f4a349b88915787fea3fb91348bb1665831bbb#diff-9478f2787e3ae9668a15123b165c23ac\">https://github.com/python-pillow/Pillow/commit/46f4a349b88915787fea3fb91348bb1665831bbb#diff-9478f2787e3ae9668a15123b165c23ac</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commits/master/src/libImaging\">https://github.com/python-pillow/Pillow/commits/master/src/libImaging</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/6.2.3.html\">https://pillow.readthedocs.io/en/stable/releasenotes/6.2.3.html</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html\">https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html</a></li>\n<li><a href=\"https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574577\">https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574577</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-2/\">https://usn.ubuntu.com/4430-2/</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/issues/4750\">https://github.com/python-pillow/Pillow/issues/4750</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-8843-m7mw-mxqm\">https://github.com/advisories/GHSA-8843-m7mw-mxqm</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-8843-m7mw-mxqm",
            "title": "[Pillow] Buffer overflow in Pillow",
            "date_modified": "2023-09-05T09:17:04.000Z",
            "date_published": "2020-07-27T21:52:41.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-8843-m7mw-mxqm"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>In <code>libImaging/Jpeg2KDecode.c</code> in Pillow before 7.1.0, there are multiple out-of-bounds reads via a crafted JP2 file.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-10994\">https://nvd.nist.gov/vuln/detail/CVE-2020-10994</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/4505\">https://github.com/python-pillow/Pillow/pull/4505</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/pull/4538\">https://github.com/python-pillow/Pillow/pull/4538</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commit/ff60894d697d1992147b791101ad53a8bf1352e4\">https://github.com/python-pillow/Pillow/commit/ff60894d697d1992147b791101ad53a8bf1352e4</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/blob/master/docs/releasenotes/7.1.0.rst#security\">https://github.com/python-pillow/Pillow/blob/master/docs/releasenotes/7.1.0.rst#security</a></li>\n<li><a href=\"https://github.com/python-pillow/Pillow/commits/master/src/libImaging/\">https://github.com/python-pillow/Pillow/commits/master/src/libImaging/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/BEBCPE4F2VHTIT6EZA2YZQZLPVDEBJGD/</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/HOKHNWV2VS5GESY7IBD237E7C6T3I427/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/\">https://pillow.readthedocs.io/en/stable/releasenotes/</a></li>\n<li><a href=\"https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html\">https://pillow.readthedocs.io/en/stable/releasenotes/7.1.0.html</a></li>\n<li><a href=\"https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574575\">https://snyk.io/vuln/SNYK-PYTHON-PILLOW-574575</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-1/\">https://usn.ubuntu.com/4430-1/</a></li>\n<li><a href=\"https://usn.ubuntu.com/4430-2/\">https://usn.ubuntu.com/4430-2/</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-vj42-xq3r-hr3r\">https://github.com/advisories/GHSA-vj42-xq3r-hr3r</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-vj42-xq3r-hr3r",
            "title": "[Pillow] Out-of-bounds reads in Pillow",
            "date_modified": "2023-09-05T09:11:45.000Z",
            "date_published": "2020-07-27T21:52:39.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-vj42-xq3r-hr3r"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>The django_ses library implements a mail backend for Django using AWS Simple Email Service. The library exports the <code>SESEventWebhookView class</code> intended to receive signed requests from AWS to handle email bounces, subscriptions, etc. These requests are signed by AWS and are verified by django_ses, however the verification of this signature was found to be flawed as it allowed users to specify arbitrary public certificates.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/django-ses/django-ses/security/advisories/GHSA-qg36-9jxh-fj25\">https://github.com/django-ses/django-ses/security/advisories/GHSA-qg36-9jxh-fj25</a></li>\n<li><a href=\"https://github.com/django-ses/django-ses/commit/b71b5f413293a13997b6e6314086cb9c22629795\">https://github.com/django-ses/django-ses/commit/b71b5f413293a13997b6e6314086cb9c22629795</a></li>\n<li><a href=\"https://github.com/django-ses/django-ses/blob/3d627067935876487f9938310d5e1fbb249a7778/CVE/001-cert-url-signature-verification.md\">https://github.com/django-ses/django-ses/blob/3d627067935876487f9938310d5e1fbb249a7778/CVE/001-cert-url-signature-verification.md</a></li>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2023-33185\">https://nvd.nist.gov/vuln/detail/CVE-2023-33185</a></li>\n<li><a href=\"https://github.com/pypa/advisory-database/tree/main/vulns/django-ses/PYSEC-2023-82.yaml\">https://github.com/pypa/advisory-database/tree/main/vulns/django-ses/PYSEC-2023-82.yaml</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-qg36-9jxh-fj25\">https://github.com/advisories/GHSA-qg36-9jxh-fj25</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-qg36-9jxh-fj25",
            "title": "[django-ses] Incorrect signature verification in django-ses",
            "date_modified": "2023-09-05T05:07:10.000Z",
            "date_published": "2023-05-22T19:41:56.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-qg36-9jxh-fj25"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<h1 id=\"server-side-template-injection\">Server-Side Template Injection</h1>\n<h2 id=\"overview-of-the-vulnerability\">Overview of the Vulnerability</h2>\n<p>Server-Side Template Injection (SSTI) is a vulnerability within application templating engines where user input is improperly handled and is embedded into the template, possibly leading code being executed.</p>\n<p>An attacker can use SSTI to execute code on the underlying system by manipulating values within the embedded template. When code is executed within the underlying system, it can allow an attacker to run permissioned commands under the exploited process, or exploit Cross-Site Scripting (XSS) to run code within the user's browser.</p>\n<h2 id=\"business-impact\">Business Impact</h2>\n<p>SSTI can lead to reputational damage for the business due to a loss in confidence and trust by users. If an attacker successfully executes code within the underlying system, it can result in data theft and indirect financial losses.</p>\n<h2 id=\"steps-to-reproduce\">Steps to Reproduce</h2>\n<ol>\n<li><a href=\"https://fief.fief.dev/register\">Sign up</a> and login to your account</li>\n<li>Use a browser to navigate to: email-templates {{<a href=\"https://test.fief.dev/admin/customization/email-templates/\">URL</a>}}</li>\n<li>put your payload in Edit Base template <code>{{ cycler.__init__.__globals__.os.popen('id').read() }}</code> and you will se it will execute.</li>\n</ol>\n<p>Payload:\n<code>{{ cycler.__init__.__globals__.os.popen('id').read() }}</code></p>\n<h2 id=\"proof-of-concept-poc\">Proof of Concept (PoC)</h2>\n<p>The screenshot(s) below demonstrates the SSTI:</p>\n<p><img alt=\"SSTI\" src=\"https://user-images.githubusercontent.com/42150485/248214990-854c2272-4f34-4c49-9759-d49ce8ce3d12.png\"></p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://github.com/fief-dev/fief/security/advisories/GHSA-hj8m-9fhf-v7jp\">https://github.com/fief-dev/fief/security/advisories/GHSA-hj8m-9fhf-v7jp</a></li>\n<li><a href=\"https://github.com/fief-dev/fief/commit/91e56625b641fa419e2985772266774bae18382b\">https://github.com/fief-dev/fief/commit/91e56625b641fa419e2985772266774bae18382b</a></li>\n<li><a href=\"https://github.com/fief-dev/fief/releases/tag/v0.25.3\">https://github.com/fief-dev/fief/releases/tag/v0.25.3</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-hj8m-9fhf-v7jp\">https://github.com/advisories/GHSA-hj8m-9fhf-v7jp</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-hj8m-9fhf-v7jp",
            "title": "[fief-server] fief-server Server-Side Template Injection vulnerability",
            "date_modified": "2023-09-05T05:06:08.000Z",
            "date_published": "2023-06-23T21:46:11.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-hj8m-9fhf-v7jp"
            },
            "tags": [
                "severity"
            ]
        },
        {
            "content_html": "<p>Python rtslib-fb through 2.1.72 has weak permissions for <code>/etc/target/saveconfig.json</code> because shutil.copyfile (instead of shutil.copy) is used, and thus permissions are not preserved.</p>\n<h3 id=\"references\">References</h3>\n<ul>\n<li><a href=\"https://nvd.nist.gov/vuln/detail/CVE-2020-14019\">https://nvd.nist.gov/vuln/detail/CVE-2020-14019</a></li>\n<li><a href=\"https://github.com/open-iscsi/rtslib-fb/pull/162\">https://github.com/open-iscsi/rtslib-fb/pull/162</a></li>\n<li><a href=\"https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TNMCV2DJJTX345YYBXAMJBXNNVUZQ5UH/\">https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/TNMCV2DJJTX345YYBXAMJBXNNVUZQ5UH/</a></li>\n<li><a href=\"http://lists.opensuse.org/opensuse-security-announce/2020-08/msg00012.html\">http://lists.opensuse.org/opensuse-security-announce/2020-08/msg00012.html</a></li>\n<li><a href=\"https://github.com/open-iscsi/rtslib-fb/commit/b23d061ee0fa7924d2cdce6194c313b9ee06c468\">https://github.com/open-iscsi/rtslib-fb/commit/b23d061ee0fa7924d2cdce6194c313b9ee06c468</a></li>\n<li><a href=\"https://github.com/advisories/GHSA-cpcw-p965-wpqx\">https://github.com/advisories/GHSA-cpcw-p965-wpqx</a></li>\n</ul>\n",
            "url": "https://github.com/advisories/GHSA-cpcw-p965-wpqx",
            "title": "[rtslib-fb] rtslib-fb weak permissions for /etc/target/saveconfig.json file",
            "date_modified": "2023-09-05T05:04:15.000Z",
            "date_published": "2022-05-24T17:21:16.000Z",
            "author": {
                "name": "GitHub",
                "url": "https://github.com/advisories/GHSA-cpcw-p965-wpqx"
            },
            "tags": [
                "severity"
            ]
        }
    ]
}